crates/worker/src/partition/mod.rs

```diff
@@ -264,6 +264,23 @@ where
 {
     #[instrument(level = "error", skip_all, fields(partition_id = %self.partition_id, is_leader = tracing::field::Empty))]
     pub async fn run(mut self) -> anyhow::Result<()> {
+        let res = self.run_inner().await;
+
+        // Drain rpc_rx
+        self.rpc_rx.close();
```

Comment:
Let's maybe do the same for the `control_rx` channel. There it is enough to simply drain and drop the messages.